### PR TITLE
Replace parsing by 'material-colors-object' module

### PR DESCRIPTION
**Pull request by amercier**
Hi @shuhei!

**TL;DR:** This PR replace its own parsing by using the [material-colors-object].

### Description

I made [material-colors-object] which parses  colors from the [Official Material Design Documentation page](https://material.io/design/color/the-color-system.html#tools-for-picking-colors). I think this package is not redundant with yours, it could actually allow separation of concerns pretty neatly:
- [material-colors-object] would parse the official documentation and expose a raw Javascript object. It needs to be updated every time the documentation format or content change.
- [material-colors] would use data from [material-colors-object] and expose `.css`, `.scss`, etc. files to coders. It wouldn't need to be updated every time the documentation format or content change.

### Notes

- The official documentation has been updated since the release of [material-colors]. URL https://material.io/guidelines/style/color.html now redirects to https://material.io/design/color/.

- The official documentation doesn't contain light and dark text information anymore. That is why [material-colors-object] doesn't expose it. This PR doesn't intent to break backward compatibility, that's these colors have been added statically in `lib/compat.json`. The only drawback is that
"compatiblity" colors are positioned after Black & White (that's what this PR contain changes in `.css`, `.scss`, etc files. See `dist/colors.json` diff for a clear picture.

- Format of [material-colors-object]'s object differs a little bit from what `fetch()` returned, so I updated the `json` grunt task in order to maintain backward compatibility.

- Official documentation now use `Gray` instead of `Grey`, so I updated the `json` grunt task in order to maintain backward compatibility.

- [material-colors-object] also contain light and dark variant for each colors (which are shown in https://material.io/tools/color/#!/). I will post another PR to add these variants in [material-colors] later.
(**Edit:** done in #23)

[material-colors-object]: https://www.npmjs.com/package/material-colors-object
[material-colors]: https://www.npmjs.com/package/material-colors

### CODE REVIEW

**Comment by shuhei** [comment]
Hi @amercier, this is great! Thanks a lot for updating the data fetching and taking care of backward compatibility.

I'll leave one minor comment about versioning. Everything else looks good to me!

File: package.json [hunk 1]
```diff
@@ -33,9 +33,9 @@
   },
   "license": "ISC",
   "devDependencies": {
-    "cheerio": "^0.22.0",
     "grunt": "^1.0.1",
     "grunt-cli": "^1.2.0",
+    "material-colors-object": "^1.0.0",
```
Comment (shuhei) [inline package.json]: I think we should have a strictly specific version `1.0.0` here because the content of `material-colors` now depends on the exact version of `material-colors-object`. Otherwise, we cannot build the same content of `material-colors` when you publish a new version of `material-colors-object` like `1.1.0` or `1.0.1` with a slightly different content.

Comment (amercier) [reply package.json]: Added 2d7266b to fix this.

**Comment by shuhei** [comment]
Thanks for updating!
